numpy/_core/src/multiarray/item_selection.c

```diff
@@ -1244,12 +1254,19 @@ _new_sortlike(PyArrayObject *op, int axis, PyArray_SortFunc *sort,
             memset(buffer, 0, N * elsize);
         }
 
-        if (swap) {
-            odescr = PyArray_DescrNewByteorder(descr, NPY_SWAP);
+        if (strided_loop != NULL) {
+            // Descriptors have already been resolved
+            odescr = context->descriptors[1];
+            Py_INCREF(odescr);
         }
         else {
-            odescr = descr;
-            Py_INCREF(odescr);
+            if (swap) {
+                odescr = PyArray_DescrNewByteorder(descr, NPY_SWAP);
+            }
+            else {
+                odescr = descr;
+                Py_INCREF(odescr);
+            }
```

Comment:
This is still slightly wrong should have explained more maybe. Needscopy needs odescr to decide. (Rather than swap since we replace the swap logic!.)
So we should find odescrfirst, and where we do, eithercheck equivtypes or check swap (or always check equivtypes later, it should take the ==fast path anyway).